pkg/argsbuilder/submit_pytorchjob.go

```diff
@@ -170,15 +172,36 @@ func (s *SubmitPytorchJobArgsBuilder) check() error {
 			return fmt.Errorf("--share-memory is invalid")
 		}
 	}
+
+	// Check whether nprocPerNode is valid
+	switch s.args.NprocPerNode {
+	case "auto", "cpu", "gpu":
+		log.Debugf("Supported nprocPerNode: %s", s.args.NprocPerNode)
+	case "":
+		log.Debugf("--nproc-per-node is not set")
+	default:
+		nprocPerNode, err := strconv.Atoi(s.args.NprocPerNode)
+		if err != nil {
+			return fmt.Errorf("--nproc-per-node is invalid")
+		}
+		log.Debugf("Supported nprocPerNode: %d", nprocPerNode)
+	}
+
 	return nil
 }
 
-func (s *SubmitPytorchJobArgsBuilder) addMasterAddr() error {
+func (s *SubmitPytorchJobArgsBuilder) addEnv() error {
+	if s.args.Envs == nil {
+		s.args.Envs = map[string]string{}
+	}
+
 	if s.args.EnableRDMA {
-		if s.args.Envs == nil {
-			s.args.Envs = map[string]string{}
-		}
 		s.args.Envs["MASTER_ADDR"] = fmt.Sprintf("%v-master-0", s.args.Name)
 	}
+
+	if s.args.NprocPerNode != "" {
```

Comment:
If the user set --env=PET_NPROC_PER_NODE=1. How to handle this logic?

Reply:
> If the user set --env=PET_NPROC_PER_NODE=1. How to handle this logic?

If the user set both `--nproc-per-node=2` and `--env=PET_NPROC_PER_NODE=1`, then there will be two environment variables named `PET_NPROC_PER_NODE` with values `1` and `2` respectively. Otherwise, there will only be one env `PET_NPROC_PER_NODE=1`.